src/components/PopupWithForm.js

```diff
@@ -8,20 +8,31 @@ export default class PopupWithForm extends Popup {
       this._popupForm.querySelectorAll(".modal__input")
     );
     this._handleFormSubmit = handleFormSubmit;
+    this._modalSubmitButton =
+      this._popupElement.querySelector(".modal__button");
   }
 
-  _setInputValues() {
-    this._formValues = {};
+  _getInputValues() {
+    const formValues = {};
     this._inputValues.forEach((input) => {
-      this._formValues[input.name] = input.value;
+      formValues[input.name] = input.value;
     });
-    return this._formValues;
+    return formValues;
   }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._modalSubmitButton.textContent = "Saving...";
+    } else {
+      this._modalSubmitButton.textContent = "Save";
+    }
+  }
+
   setEventListeners() {
-    super.setEventListiners();
+    super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._setInputValues());
+      this._handleFormSubmit(this._getInputValues());
       this.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
* One remark hasn't been corrected https://github.com/azatmanas/se_project_aroundtheus/pull/3/files#r1751189954